src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -12,12 +12,24 @@
 'use strict';
 
 var ReactOwner = require('ReactOwner');
+var warning = require('warning');
 
 var ReactRef = {};
 
 function attachRef(ref, component, owner) {
   if (typeof ref === 'function') {
-    ref(component.getPublicInstance());
+    var instance = component.getPublicInstance();
+    if (__DEV__) {
+      var componentName = component && component.getName ?
+        component.getName() : 'a component';
+      warning(instance != null,
+        'Stateless function components cannot be given refs ' +
+        '(See %s%s).',
+        componentName,
+        owner ? ' created by ' + owner.getName() : ''
```

Comment:
IMO this should follow the normal pattern `check the render method of ...`
Something like `Stateless functional component %s cannot have refs attached. Check the render method of %s`. I also think that the other message (when using a string ref) should follow this pattern too.